starlette/routing.py

```diff
@@ -647,6 +651,19 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
 
         partial = None
 
+        if self.user_ctx is not None:
+            current_ctx = contextvars.copy_context()
+            for cvar in self.user_ctx:
+                if cvar in current_ctx:
+                    # not set by the user, skip to avoid modifying
+                    # vars used by event loops, servers, etc.
+                    continue
```

Comment:
@graingert note that without this check, I was running into issues with Trio (not asyncio) here: https://github.com/python-trio/trio/blob/2865222187a6b51f7a1df54a8c4445574c86183d/trio/_core/_run.py#L2296

It seems that copying back the context made Trio think that it was in the lifespan task permanently or something.